// material-ui-modal/src/material-ui-alert-modal/AlertModalProvider.js
import React from 'react';
import { AlertModal } from './AlertModal';
const ModalContext = React.createContext();
const defaultOptions = {
    anchorOrigin: { vertical: 'top', horizontal: 'center' },
    autoHideDuration: 3000,
    severity: 'info',
    title: 'default title',
    message: 'default message'
}

const buildOptions = (options) => {
    return Object.assign({}, defaultOptions, options);
}

const AlertModalProvider = (props) => {
    const [options, setOptions] = React.useState(defaultOptions);
    const [open, setOpen] = React.useState(false);

    const handleClose = () => {
        setOpen(false);
    }

    const showModal = React.useCallback(options => {
        setOpen(true);
        setOptions(buildOptions(options));
    }, [])

    return (
        <React.Fragment>
            <ModalContext.Provider value={showModal}>
                {props.children}
            </ModalContext.Provider>
            <AlertModal options={options} open={open} onClose={handleClose} />
        </React.Fragment>
    )
}

const useAlertModal = () => {
    var modal = React.useContext(ModalContext);
    return modal;
};

export { AlertModalProvider, useAlertModal }
